### PR TITLE
Native MacOSX games Support

### PR DESCRIPTION
**Pull request by flavioislima**
This Adds several handlers to deal with MacOSX native games:

- [x] Add type `is_mac_native`to differentiate between Mac and Windows games.
- [x] Add filter on the Library for platforms (only on Mac)
- [x] Deal with launching native games
- [x] Show Platform information on InstallModal
- [x] Remove unnecessary settings for native games (wine, show fps, etc)
- [x] Show correct platform for already windows installed games that have Mac support as well.

Screenshot 2021-12-20 at 21 25 21

---

Use the following Checklist if you have changed something on the Backend or Frontend:

- [x] Tested the feature and it's working on a current and clean install.
- [x] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [x] Created / Updated Tests (If necessary)
- [x] Created / Updated documentation (If necessary)

### CODE REVIEW

File: src/screens/Settings/components/OtherSettings/index.test.tsx [hunk 1]
```diff
@@ -32,6 +32,7 @@ interface Props {
   toggleDiscordRPC: () => void
   useGameMode: boolean
   targetExe: string
+  isNativeMac: boolean
```
Comment (Nocccer) [inline src/screens/Settings/components/OtherSettings/index.test.tsx:35]: sometimes you call it "isMacNative" here you call it "isNativeMac". Is this on porpuse ?

File: src/screens/Settings/components/OtherSettings/index.tsx [hunk 2]
```diff
@@ -15,6 +15,7 @@ const { ipcRenderer } = window.require('electron') as {
 interface Props {
   audioFix: boolean
   isDefault: boolean
+  isNativeMac: boolean
```
Comment (Nocccer) [inline src/screens/Settings/components/OtherSettings/index.tsx:18]: Same here with "isNativeMac" and "isMacNative"

File: src/screens/Settings/index.tsx [hunk 3]
```diff
@@ -207,9 +211,11 @@ function Settings() {
         const {
           cloud_save_enabled: cloudSaveEnabled,
           save_folder: saveFolder,
-          title: gameTitle
+          title: gameTitle,
+          is_mac_native
         } = await getGameInfo(appName)
         setTitle(gameTitle)
+        setIsNativeMac(is_mac_native)
```
Comment (Nocccer) [inline src/screens/Settings/index.tsx:218]: Here is a good example for the variable spelling. Is this on porpuse ?

**Review by Nocccer — CHANGES_REQUESTED** [review]
Open question left. Else LGTM

File: src/screens/Settings/index.tsx (hunk 3 above)
Comment (flavioislima) [reply src/screens/Settings/index.tsx:218]: `is_mac_native` is a field from the library so belongs to the GameInfo. So to follow the standard we have today I decided to keep like this.

Comment (Nocccer) [reply src/screens/Settings/index.tsx:218]: I would love to have setIsMacNative. So it follows the spelling of is_mac_native.

**Review by Nocccer — APPROVED** [review]
LGTM

**Pull request merged**